src/xcode/ENA/ENA/Source/Scenes/ContactDiary/AddAndEditEntry/DiaryAddAndEditEntryViewController.swift

```diff
@@ -73,34 +78,41 @@ class DiaryAddAndEditEntryViewController: UIViewController, UITextFieldDelegate,
 	}
 
 	func textFieldShouldReturn(_ textField: UITextField) -> Bool {
-		entryTextField.resignFirstResponder()
-		viewModel.save()
-		entryTextField.resignFirstResponder()
-		dismiss()
+		switch textField.returnKeyType {
+		case .default, .done, .send:
+			if !viewModel.entryModel.isEmpty {
+				viewModel.save()
+			}
+			textfieldsManager.resignFirstResponder()
+			dismiss()
+		case .next, .continue:
+			textfieldsManager.nextFirstResponder()
+		default:
+			Log.debug("unsupport return key type")
```

Comment:
Should we call textfieldsManager.nextFirstResponder() here?

Reply:
no, I think the default behavior should be to enable next first responder only on .next or .continue. But we are open here to change things later if needed